framework/ui/src/components/molecules/vc-select/vc-select.vue

```diff
@@ -28,9 +28,18 @@
         <div v-if="!selectedItem" class="vc-select__field-placeholder">
           {{ placeholder }}
         </div>
-        <slot v-else name="item" :item="selectedItem">
+        <slot
+          v-else-if="selectedItem && !customSelectedItem"
+          name="item"
+          :item="selectedItem"
+        >
           {{ selectedItem[displayProperty] }}
         </slot>
+        <slot
+          v-else-if="customSelectedItem"
+          name="customItem"
```

Comment:
Rename to `selectedItem`, and use `$slots['selectedItem']` check to descide if you need to show it or default `item` slot instead of `customSelectedItem` property.
So, if `selectedItem` slot is passed, then use it, else use `item` slot. This is enough.